lib/srv/db/redis/cmds.go

```diff
@@ -177,6 +177,16 @@ func (e *Engine) processAuth(ctx context.Context, cmd *redis.Cmd) error {
 			return trace.Wrap(err)
 		}
 
+		password, ok := cmd.Args()[1].(string)
+		if !ok {
+			return trace.BadParameter("password has a wrong type, expected string got %T", cmd.Args()[1])
+		}
+
+		e.redisClient, err = e.reconnect(defaults.DefaultRedisUsername, password)
```

Comment:
Should we use empty `""` here for username so the client sends `AUTH <password>` instead of `AUTH <user> <password>`?

Reply:
I think we should still send the username as a part of `AUTH` regardless. I know that AWS allows you to change the default username (and may be not the only one). Currently if you skip the `--db-user` in `tsh` Teleport assumes that you're the `default` user and we're using that name for our RBAC check. If for some reason Redis has different user configured as the default, then we may allow access when we really shouldn't. 
In term of backward compatibility we should be fine as we already require Redis 6 anyways.

Reply:
I couldn't find where ElastiCache can change the `default` username. It allows swapping a different default ElastiCache user in the user group but it must have the same in-database name `default`. Could you share if you have a link to how to change `default`?
Any other reason we limit to Redis 6?  I tried ElastiCache with older version seems working fine.

Reply:
@greedy52 You're right. The `default` user can be only swapped with another `default` user. I changed the code to use in this case empty username.
> Any other reason we limit to Redis 6? I tried ElastiCache with older version seems working fine.

The reason why we require Redis 6 for standalone version is that Redis 6 added TLS and ACL. I know that AWS allows you to use TLS with older Redis than 6, but it still lacks ACL.

@r0mant @smallinsky Do you think that for AWS we could lower required Redis version as AWS supports TLS starting from Redis 3/4?
https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/in-transit-encryption.html

> In-transit encryption is supported on replication groups running Redis versions 3.2.6, 4.0.10 and later.

Reply:
in current implementation for ElastiCache, support is based on TLS enabled or not. We can cerntainly add version check later if necessary.
https://github.com/gravitational/teleport/blob/64b86754f935260b190b94ea34e2e9f4925ec635/lib/services/database.go#L537-L541

Reply:
> Do you think that for AWS we could lower required Redis version as AWS supports TLS starting from Redis 3/4?
Yes if TLS is supported but ACL is missing we can support support this case  👍